cache_path.go

```diff
@@ -191,17 +192,25 @@ func normalizeExcludeByPattern(excludeByPattern map[string]bool) (map[string]boo
 
 // match reports whether the path matches to any of the given ignore items
 // and returns the exclude property of the matching ignore item.
-func match(pth string, excludeByPattern map[string]bool) (bool, bool) {
-	for pattern, exclude := range excludeByPattern {
-		if strings.Contains(pattern, "*") && glob.Glob(pattern, pth) {
-			return true, exclude
+func match(pth string, excludeByPattern map[string]bool) (exclude bool, ok bool) {
+	matchFn := func(patternOrPath, subject string) bool {
```

Comment:
this would be better to be moved out. feels strange reading matchFn with type function inside the function match

for example `isPatternMatching` or something like that.